Game/Entities/Player.cs

```diff
@@ -33,6 +33,21 @@ public override void _PhysicsProcess(double delta)
 
 		Velocity = velocity;
 		MoveAndSlide();
+
+		//controller joystick controls
+		Vector2 JoystickDirection = Input.GetVector("joystick_left", "joystick_right", "joystick_up", "joystick_down");
+		if (JoystickDirection != Vector2.Zero)
+		{
+			velocity = JoystickDirection.Normalized() * Speed;
+		}
+		else
+		{
+			velocity.X = Mathf.MoveToward(Velocity.X, 0, Speed);
+			velocity.Y = Mathf.MoveToward(Velocity.Y, 0, Speed);
+		}
+
+		Velocity = velocity;
+		MoveAndSlide();
```

Comment:
Probably don't wanna do this twice, maybe combine the joystick and keyboard vectors into one and only do the movement stuff once

Reply:
`
Vector2 direction = Input.GetVector("ui_left", "ui_right", "ui_up", "ui_down");
Vector2 JoystickDirection = Input.GetVector("joystick_left", "joystick_right", "joystick_up", "joystick_down");
var resultVec = direction + JoystickDirection;
// clamp vector using Vector.clamp() and then continue as normal
`